config/src/main/java/org/springframework/security/config/annotation/web/configuration/ConversionServicePostProcessor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.config.annotation.web.configuration;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.convert.converter.ConverterRegistry;
+import org.springframework.core.convert.support.ConfigurableConversionService;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.security.crypto.factory.RsaKeyConverters;
+
+/**
+ * Adds Spring Security converters to the configured {@link ConversionService}
+ *
+ * @author Josh Cummings
+ * @since 5.2
+ */
+public class ConversionServicePostProcessor implements BeanFactoryPostProcessor {
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) throws BeansException {
+		ConversionService service = beanFactory.getConversionService();
+		if (service instanceof ConfigurableConversionService) {
+			addRsaConverters((ConfigurableConversionService) service);
+		}
+	}
+
+	private void addRsaConverters(ConverterRegistry converterRegistry) {
+		Converter<String, InputStream> pemInputStreamConverter = pemInputStreamConverter();
+		converterRegistry.addConverter(String.class, RSAPrivateKey.class,
+				pair(pemInputStreamConverter, RsaKeyConverters.pkcs8()));
+		converterRegistry.addConverter(String.class, RSAPublicKey.class,
+				pair(pemInputStreamConverter, RsaKeyConverters.x509()));
+	}
+
+	private Converter<String, InputStream> pemInputStreamConverter() {
+		ResourceLoader resourceLoader = new DefaultResourceLoader();
+		return source -> source.startsWith("-----") ?
```

Comment:
Can you help me understand what the conditional logic is for? It looks like this might be to handle things like `classpath:foo.pem`. However, my concern is that it should not interfere with existing String to `InputStream` conversions.

Reply:
Does this work if a variable (i.e. `${some.path}` is used)? I cannot recall if anything special is necessary, but it has bitten me before.

Reply:
Regarding your first question, the actual registered converter is `String -> RSAPublicKey` so it will only be invoked when the target type is `RSAPublicKey`. Yes, the conditional logic is to differentiate between a raw key and one referred to via `classpath:foo.pem`.

I didn't test your second case, but I will double-check this.

Reply:
I changed the sample to use `${some.property.name}` to confirm that works.